modules/basic/src/main/java/com/devonfw/module/basic/common/api/AdvancedRevisionEntity.java

```diff
@@ -0,0 +1,124 @@
+package com.devonfw.module.basic.common.api;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.hibernate.envers.RevisionEntity;
+import org.hibernate.envers.RevisionNumber;
+import org.hibernate.envers.RevisionTimestamp;
+
+import com.devonfw.module.basic.common.api.entity.PersistenceEntity;
+
+/**
+ * This is a custom {@link org.hibernate.envers.DefaultRevisionEntity revision entity} also containing the actual user.
+ *
+ * @see org.hibernate.envers.DefaultRevisionEntity
+ */
+@Entity
+@RevisionEntity(AdvancedRevisionListener.class)
+@Table(name = "RevInfo")
+public class AdvancedRevisionEntity implements PersistenceEntity<Long> {
```

Comment:
Can you give a rationale why we had to move the Entity classes to common?
This looks odd to me. We had this once in OASP4J long time ago and got lots of user complaints from the fact that they were using other Databases and did not have control over `@GeneratedValue`, etc. or the annotation style (private field vs. getter annotation). Also devon4j with oasp4j has a long track record of private field vs. getter annotation for JPA. Chaning this would require documentation updates and some discussion.

My suggestion would be to only add the `RevisionMetadata` with its `RevisionMetadataType` implementation here and leave everything else as is.
IMHO you also need to check compatibilitiy / deprecation for existing `RevisionMetadataType` and `LazyRevisionMetadata`. Whenever we do some deprecation like this, it has to be consistent and ensure that:
* all existing types remain in the same place and get `@deprecated`
* for everything we introduce a replacement as a copy that is not coupled to the `deprecated` original. E.g. the existing `LazyRevisionMetadata` still has to implement the old `deprecated` interface as otherwise we would break compatibilitiy. This way it is therefore forever coupled with the deprecated world. We need to introduce a new implementation that only implements the new interface without being connected with the `deprecated` code.